### PR TITLE
Add Gradient Accumulation Option

### PR DESCRIPTION
**Pull request by jstremme**
Adds the option to accumulate gradients and update model weights every `accumulation_steps` steps.  This enables training at a larger effective batch size than what can fit into memory.  For large models or when GPU RAM is scarce, this can help maintain good estimates of the gradient (by using more samples to estimate the gradient) without encountering memory errors.

Thanks btw for the great repo!  I used it in [this paper](https://proceedings.mlr.press/v193/stremmel22a.html) (thanks for the shoutout in your README!) and continue to build on top of it for other work.

### CODE REVIEW

**Comment by MichalBrzozowski91** [comment]
Thank you, I am glad that you enjoy the repo. The gradient accumulation is a very useful addition. Is the PR ready for review?

**Comment by jstremme** [comment]
> Thank you, I am glad that you enjoy the repo. The gradient accumulation is a very useful addition. Is the PR ready for review?

Yes, ready for review.  Just tested the example notebook and the model trains without issue using gradient accumulation.

File: belt_nlp/bert.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def __init__(
         batch_size: int,
         learning_rate: float,
         epochs: int,
+        accumulation_steps: Optional[int] = 1,
```
Comment (MichalBrzozowski91) [inline belt_nlp/bert.py:36]: Do we need Optional here and in other places?

Comment (jstremme) [reply belt_nlp/bert.py:36]: I think so.  We just want this to default to 1 in case it is not passed by the user.  Accumulation steps = 1 is equivalent to no gradient accumulation.

Comment (MichalBrzozowski91) [reply belt_nlp/bert.py:36]: Optional typing hint allows for the `None` value. I suggest leaving just `int` type, The same way it is type hinted in lightning: https://github.com/Lightning-AI/pytorch-lightning/blob/7cc79fe7bad6b6faac0a7889ad81c4050bc32c86/src/lightning/pytorch/trainer/trainer.py#L121

Comment (jstremme) [reply belt_nlp/bert.py]: Ah, makes sense - thank you!

**Comment by MichalBrzozowski91** [comment]
Great job, thank you!
